src/lib_json/json_writer.cpp

```diff
@@ -159,7 +159,8 @@ String valueToString(double value, bool useSpecialFloats,
 
   // try to ensure we preserve the fact that this was given to us as a double on
   // input
-  if (buffer.find('.') == buffer.npos && buffer.find('e') == buffer.npos) {
+  if (buffer.find('.') == Json::String::npos &&
```

Comment:
Downgrade. `npos` is a property of the object you just did a `find` on. If the string type of that object changes, then the particular `npos` you compare the result with should follow.